### DISCUSSION TITLE
Needleman-Wunsch algorithm in BORIS?

### DISCUSSION THREAD

**Discussion by Kathahara** [olivierfriard/BORIS#611] [Q&A]
Hi, I am interested in using the Needleman-Wunsch function in BORIS - I assume temporal NW after Syed & Das 2015?
I cannot see or change any of the settings except the time unit though. How is the scoring system set by default? How are gaps scored? In my case, I would interpret gaps as NAs between two time sequences and would like to give it a score of 0. Please help me understand! If it works, it would be a great tool for me to use! Best, Kathi

**Comment by olivierfriard** [comment]
Hi,
The coded behaviors of the two observations are concatenated (without taking into account the duration of state events) in two behavioral sequences. These behavioral sequences are then submitted to the Needleman-Wunsch algorithm with the following parameters:
    match_award = 1
    mismatch_penalty = -1
    gap_penalty = -1

For more details check the needleman_wunsch_identity function in https://github.com/olivierfriard/BORIS/blob/master/boris/irr.py

**Reply by Kathahara** [reply]
Thanks for quickly getting back to me! 
I work with camera-trap data that has far from perfect videotaping qualities, i.e. I have a lot of NAs in the dataset. Do you think it would be possible in my case to set the gap_penalty to 0 accordingly? I mean since I'm not interested in the gaps but really just the similarities. 
Also, I am not sure what you mean by not taking the duration into account. Despite the NAs, I have to fixed intervals in a fixed video length (pre- and post playback) that would be a crucial part to account for. I wonder how suitable you consider BORIS's needleman-wunsch option knowing more about my aims.
